src/routes/safe/store/selectors/index.js

```diff
@@ -161,14 +161,12 @@ export type SafeSelectorProps = Safe | typeof undefined
 export const safeSelector: Selector<GlobalState, RouterProps, SafeSelectorProps> = createSelector(
   safesMapSelector,
   safeParamAddressFromStateSelector,
-  (safes: Map<string, Safe>, address: string) => {
+  (safes: Map<string, Safe>, address: string | null) => {
     if (!address) {
       return undefined
     }
-    const checksumed = getWeb3().utils.toChecksumAddress(address)
```

Comment:
In fact, @mikheevm, there was a solution in place. Just updated as `safeParamAddressFromStateSelector` is returning the checksummed address. 

So, the deletion was already working well (just tested on `development`)